// @indexcoop/tokenlists/build/cjs/index.js
"use strict";
/**
 * This is the index of the project. Default export from this file
 * can be imported from the published npm package like so:
 *
 * ```js
 * import PackageName from '@indexcoop/tokenlists';
 * ```
 *
 * And non-default exports can be imported like this:
 *
 * ```jss
 * import {functionName} from '@indexcoop/tokenlists';
 * ```
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexCoopOptimismTokens = exports.OptimismTokens = exports.IndexCoopMaticTokens = exports.MaticTokens = exports.IndexCoopMainnetTokens = exports.MainnetTokens = exports.IndexCoopAvalancheTokens = exports.AvalancheTokens = exports.IndexCoopArbitrumTokens = exports.ArbitrumTokens = void 0;
const arbitrum_1 = __importStar(require("./lists/arbitrum"));
const avalanche_1 = __importStar(require("./lists/avalanche"));
const mainnet_1 = __importStar(require("./lists/mainnet"));
const matic_1 = __importStar(require("./lists/matic"));
const optimism_1 = __importStar(require("./lists/optimism"));
exports.ArbitrumTokens = arbitrum_1.default;
exports.IndexCoopArbitrumTokens = arbitrum_1.INDEX_COOP_ARBITRUM_TOKENS;
exports.AvalancheTokens = avalanche_1.default;
exports.IndexCoopAvalancheTokens = avalanche_1.INDEX_COOP_AVALANCHE_TOKENS;
exports.MainnetTokens = mainnet_1.default;
exports.IndexCoopMainnetTokens = mainnet_1.INDEX_COOP_MAINNET_TOKENS;
exports.MaticTokens = matic_1.default;
exports.IndexCoopMaticTokens = matic_1.INDEX_COOP_MATIC_TOKENS;
exports.OptimismTokens = optimism_1.default;
exports.IndexCoopOptimismTokens = optimism_1.INDEX_COOP_OPTIMISM_TOKENS;
