// @setprotocol/set-protocol-v2/dist/utils/contracts/curve.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stableswap = exports.Minter = exports.LiquidityGaugeReward = exports.LiquidityGauge = exports.GaugeController = exports.CRVToken = exports.CurvePoolERC20 = exports.CurveDeposit = void 0;
// External Curve Contracts
var CurveDeposit_1 = require("../../typechain/CurveDeposit");
Object.defineProperty(exports, "CurveDeposit", { enumerable: true, get: function () { return CurveDeposit_1.CurveDeposit; } });
var CurvePoolERC20_1 = require("../../typechain/CurvePoolERC20");
Object.defineProperty(exports, "CurvePoolERC20", { enumerable: true, get: function () { return CurvePoolERC20_1.CurvePoolERC20; } });
var CRVToken_1 = require("../../typechain/CRVToken");
Object.defineProperty(exports, "CRVToken", { enumerable: true, get: function () { return CRVToken_1.CRVToken; } });
var GaugeController_1 = require("../../typechain/GaugeController");
Object.defineProperty(exports, "GaugeController", { enumerable: true, get: function () { return GaugeController_1.GaugeController; } });
var LiquidityGauge_1 = require("../../typechain/LiquidityGauge");
Object.defineProperty(exports, "LiquidityGauge", { enumerable: true, get: function () { return LiquidityGauge_1.LiquidityGauge; } });
var LiquidityGaugeReward_1 = require("../../typechain/LiquidityGaugeReward");
Object.defineProperty(exports, "LiquidityGaugeReward", { enumerable: true, get: function () { return LiquidityGaugeReward_1.LiquidityGaugeReward; } });
var Minter_1 = require("../../typechain/Minter");
Object.defineProperty(exports, "Minter", { enumerable: true, get: function () { return Minter_1.Minter; } });
var Stableswap_1 = require("../../typechain/Stableswap");
Object.defineProperty(exports, "Stableswap", { enumerable: true, get: function () { return Stableswap_1.Stableswap; } });
