// @setprotocol/set-protocol-v2/dist/utils/contracts/uniswap.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniswapV2Router02 = exports.UniswapV2Pair = exports.UniswapV2Factory = exports.UniswapTimelock = exports.UniswapGovernorAlpha = exports.Uni = exports.StakingRewards = void 0;
// External Uniswap Contracts
var StakingRewards_1 = require("../../typechain/StakingRewards");
Object.defineProperty(exports, "StakingRewards", { enumerable: true, get: function () { return StakingRewards_1.StakingRewards; } });
var Uni_1 = require("../../typechain/Uni");
Object.defineProperty(exports, "Uni", { enumerable: true, get: function () { return Uni_1.Uni; } });
var UniswapGovernorAlpha_1 = require("../../typechain/UniswapGovernorAlpha");
Object.defineProperty(exports, "UniswapGovernorAlpha", { enumerable: true, get: function () { return UniswapGovernorAlpha_1.UniswapGovernorAlpha; } });
var UniswapTimelock_1 = require("../../typechain/UniswapTimelock");
Object.defineProperty(exports, "UniswapTimelock", { enumerable: true, get: function () { return UniswapTimelock_1.UniswapTimelock; } });
var UniswapV2Factory_1 = require("../../typechain/UniswapV2Factory");
Object.defineProperty(exports, "UniswapV2Factory", { enumerable: true, get: function () { return UniswapV2Factory_1.UniswapV2Factory; } });
var UniswapV2Pair_1 = require("../../typechain/UniswapV2Pair");
Object.defineProperty(exports, "UniswapV2Pair", { enumerable: true, get: function () { return UniswapV2Pair_1.UniswapV2Pair; } });
var UniswapV2Router02_1 = require("../../typechain/UniswapV2Router02");
Object.defineProperty(exports, "UniswapV2Router02", { enumerable: true, get: function () { return UniswapV2Router02_1.UniswapV2Router02; } });
