// @setprotocol/set-protocol-v2/dist/utils/deploys/deployAdapters.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var AaveGovernanceAdapter__factory_1 = require("../../typechain/factories/AaveGovernanceAdapter__factory");
var AaveGovernanceV2Adapter__factory_1 = require("../../typechain/factories/AaveGovernanceV2Adapter__factory");
var AaveV2WrapV2Adapter__factory_1 = require("../../typechain/factories/AaveV2WrapV2Adapter__factory");
var AxieInfinityMigrationWrapAdapter__factory_1 = require("../../typechain/factories/AxieInfinityMigrationWrapAdapter__factory");
var BalancerV1IndexExchangeAdapter__factory_1 = require("../../typechain/factories/BalancerV1IndexExchangeAdapter__factory");
var CompoundLikeGovernanceAdapter__factory_1 = require("../../typechain/factories/CompoundLikeGovernanceAdapter__factory");
var CurveStakingAdapter__factory_1 = require("../../typechain/factories/CurveStakingAdapter__factory");
var KyberExchangeAdapter__factory_1 = require("../../typechain/factories/KyberExchangeAdapter__factory");
var KyberV3IndexExchangeAdapter__factory_1 = require("../../typechain/factories/KyberV3IndexExchangeAdapter__factory");
var KyberMigrationWrapAdapter__factory_1 = require("../../typechain/factories/KyberMigrationWrapAdapter__factory");
var OneInchExchangeAdapter__factory_1 = require("../../typechain/factories/OneInchExchangeAdapter__factory");
var ZeroExApiAdapter__factory_1 = require("../../typechain/factories/ZeroExApiAdapter__factory");
var AaveMigrationWrapAdapter__factory_1 = require("../../typechain/factories/AaveMigrationWrapAdapter__factory");
var AaveWrapAdapter__factory_1 = require("../../typechain/factories/AaveWrapAdapter__factory");
var CompoundWrapAdapter__factory_1 = require("../../typechain/factories/CompoundWrapAdapter__factory");
var CompoundWrapV2Adapter__factory_1 = require("../../typechain/factories/CompoundWrapV2Adapter__factory");
var YearnWrapAdapter__factory_1 = require("../../typechain/factories/YearnWrapAdapter__factory");
var YearnWrapV2Adapter__factory_1 = require("../../typechain/factories/YearnWrapV2Adapter__factory");
var UniswapPairPriceAdapter__factory_1 = require("../../typechain/factories/UniswapPairPriceAdapter__factory");
var UniswapV2ExchangeAdapter__factory_1 = require("../../typechain/factories/UniswapV2ExchangeAdapter__factory");
var UniswapV2AmmAdapter__factory_1 = require("../../typechain/factories/UniswapV2AmmAdapter__factory");
var UniswapV2TransferFeeExchangeAdapter__factory_1 = require("../../typechain/factories/UniswapV2TransferFeeExchangeAdapter__factory");
var UniswapV2ExchangeAdapterV2__factory_1 = require("../../typechain/factories/UniswapV2ExchangeAdapterV2__factory");
var UniswapV2IndexExchangeAdapter__factory_1 = require("../../typechain/factories/UniswapV2IndexExchangeAdapter__factory");
var UniswapV3IndexExchangeAdapter__factory_1 = require("../../typechain/factories/UniswapV3IndexExchangeAdapter__factory");
var UniswapV3ExchangeAdapter__factory_1 = require("../../typechain/factories/UniswapV3ExchangeAdapter__factory");
var SnapshotGovernanceAdapter__factory_1 = require("../../typechain/factories/SnapshotGovernanceAdapter__factory");
var SynthetixExchangeAdapter__factory_1 = require("../../typechain/factories/SynthetixExchangeAdapter__factory");
var CompoundBravoGovernanceAdapter__factory_1 = require("../../typechain/factories/CompoundBravoGovernanceAdapter__factory");
var typechain_1 = require("../../typechain");
var DeployAdapters = /** @class */ (function () {
    function DeployAdapters(deployerSigner) {
        this._deployerSigner = deployerSigner;
    }
    DeployAdapters.prototype.deployKyberExchangeAdapter = function (kyberNetworkProxy) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new KyberExchangeAdapter__factory_1.KyberExchangeAdapter__factory(this._deployerSigner).deploy(kyberNetworkProxy)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployOneInchExchangeAdapter = function (approveAddress, exchangeAddress, swapFunctionSignature) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new OneInchExchangeAdapter__factory_1.OneInchExchangeAdapter__factory(this._deployerSigner).deploy(approveAddress, exchangeAddress, swapFunctionSignature)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV2AmmAdapter = function (uniswapV2Router) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2AmmAdapter__factory_1.UniswapV2AmmAdapter__factory(this._deployerSigner).deploy(uniswapV2Router)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV2ExchangeAdapter = function (uniswapV2Router) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2ExchangeAdapter__factory_1.UniswapV2ExchangeAdapter__factory(this._deployerSigner).deploy(uniswapV2Router)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV2TransferFeeExchangeAdapter = function (uniswapV2Router) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2TransferFeeExchangeAdapter__factory_1.UniswapV2TransferFeeExchangeAdapter__factory(this._deployerSigner).deploy(uniswapV2Router)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV2ExchangeAdapterV2 = function (uniswapV2Router) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2ExchangeAdapterV2__factory_1.UniswapV2ExchangeAdapterV2__factory(this._deployerSigner).deploy(uniswapV2Router)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV2IndexExchangeAdapter = function (uniswapV2Router) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2IndexExchangeAdapter__factory_1.UniswapV2IndexExchangeAdapter__factory(this._deployerSigner).deploy(uniswapV2Router)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAaveGovernanceAdapter = function (aaveProtoGovernance, aaveToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveGovernanceAdapter__factory_1.AaveGovernanceAdapter__factory(this._deployerSigner).deploy(aaveProtoGovernance, aaveToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAaveGovernanceV2Adapter = function (aaveGovernanceV2, aaveToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveGovernanceV2Adapter__factory_1.AaveGovernanceV2Adapter__factory(this._deployerSigner).deploy(aaveGovernanceV2, aaveToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAaveMigrationWrapAdapter = function (aaveMigrationProxy, lendToken, aaveToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveMigrationWrapAdapter__factory_1.AaveMigrationWrapAdapter__factory(this._deployerSigner).deploy(aaveMigrationProxy, lendToken, aaveToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployKyberMigrationWrapAdapter = function (kncLegacyToken, kncToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new KyberMigrationWrapAdapter__factory_1.KyberMigrationWrapAdapter__factory(this._deployerSigner).deploy(kncLegacyToken, kncToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAaveWrapAdapter = function (aaveLendingPool) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveWrapAdapter__factory_1.AaveWrapAdapter__factory(this._deployerSigner).deploy(aaveLendingPool)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAxieInfinityMigrationWrapAdapter = function (tokenSwap, oldToken, newToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AxieInfinityMigrationWrapAdapter__factory_1.AxieInfinityMigrationWrapAdapter__factory(this._deployerSigner).deploy(tokenSwap, oldToken, newToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAGIMigrationWrapAdapter = function (agiLegacyToken, agixToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new typechain_1.AGIMigrationWrapAdapter__factory(this._deployerSigner).deploy(agiLegacyToken, agixToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployCompoundWrapAdapter = function (libraryName, libraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new CompoundWrapAdapter__factory_1.CompoundWrapAdapter__factory((_a = {},
                            _a[libraryName] = libraryAddress,
                            _a), this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployCompClaimAdapter = function (comptrollerAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new typechain_1.CompClaimAdapter__factory(this._deployerSigner).deploy(comptrollerAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployYearnWrapAdapter = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new YearnWrapAdapter__factory_1.YearnWrapAdapter__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployBalancerV1IndexExchangeAdapter = function (balancerProxy) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new BalancerV1IndexExchangeAdapter__factory_1.BalancerV1IndexExchangeAdapter__factory(this._deployerSigner).deploy(balancerProxy)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployCompoundLikeGovernanceAdapter = function (governanceAlpha, governanceToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundLikeGovernanceAdapter__factory_1.CompoundLikeGovernanceAdapter__factory(this._deployerSigner).deploy(governanceAlpha, governanceToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployCompoundBravoGovernanceAdapter = function (governorBravo, governanceToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundBravoGovernanceAdapter__factory_1.CompoundBravoGovernanceAdapter__factory(this._deployerSigner).deploy(governorBravo, governanceToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployCurveStakingAdapter = function (gaugeController) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CurveStakingAdapter__factory_1.CurveStakingAdapter__factory(this._deployerSigner).deploy(gaugeController)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapPairPriceAdapter = function (controller, uniswapFactory, uniswapPools) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapPairPriceAdapter__factory_1.UniswapPairPriceAdapter__factory(this._deployerSigner).deploy(controller, uniswapFactory, uniswapPools)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.getUniswapPairPriceAdapter = function (uniswapAdapterAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapPairPriceAdapter__factory_1.UniswapPairPriceAdapter__factory(this._deployerSigner).attach(uniswapAdapterAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV3IndexExchangeAdapter = function (router) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV3IndexExchangeAdapter__factory_1.UniswapV3IndexExchangeAdapter__factory(this._deployerSigner).deploy(router)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployZeroExApiAdapter = function (zeroExAddress, wethAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new ZeroExApiAdapter__factory_1.ZeroExApiAdapter__factory(this._deployerSigner).deploy(zeroExAddress, wethAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deploySnapshotGovernanceAdapter = function (delegateRegistry) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new SnapshotGovernanceAdapter__factory_1.SnapshotGovernanceAdapter__factory(this._deployerSigner).deploy(delegateRegistry)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deploySynthetixExchangeAdapter = function (synthetixExchangerAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new SynthetixExchangeAdapter__factory_1.SynthetixExchangeAdapter__factory(this._deployerSigner).deploy(synthetixExchangerAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployUniswapV3ExchangeAdapter = function (swapRouter) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV3ExchangeAdapter__factory_1.UniswapV3ExchangeAdapter__factory(this._deployerSigner).deploy(swapRouter)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployKyberV3IndexExchangeAdapter = function (dmmRouter, dmmFactory) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new KyberV3IndexExchangeAdapter__factory_1.KyberV3IndexExchangeAdapter__factory(this._deployerSigner).deploy(dmmRouter, dmmFactory)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployCompoundWrapV2Adapter = function (libraryName, libraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new CompoundWrapV2Adapter__factory_1.CompoundWrapV2Adapter__factory((_a = {},
                            _a[libraryName] = libraryAddress,
                            _a), this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployYearnWrapV2Adapter = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new YearnWrapV2Adapter__factory_1.YearnWrapV2Adapter__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployAdapters.prototype.deployAaveV2WrapV2Adapter = function (lendingPool) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2WrapV2Adapter__factory_1.AaveV2WrapV2Adapter__factory(this._deployerSigner).deploy(lendingPool)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return DeployAdapters;
}());
exports.default = DeployAdapters;
