// @setprotocol/set-protocol-v2/dist/utils/deploys/deployExternal.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("../common");
var CERc20__factory_1 = require("../../typechain/factories/CERc20__factory");
var CEther__factory_1 = require("../../typechain/factories/CEther__factory");
var CompoundPriceOracleMock__factory_1 = require("../../typechain/factories/CompoundPriceOracleMock__factory");
var Comp__factory_1 = require("../../typechain/factories/Comp__factory");
var CompoundGovernorAlpha__factory_1 = require("../../typechain/factories/CompoundGovernorAlpha__factory");
var CompoundGovernorBravoDelegator__factory_1 = require("../../typechain/factories/CompoundGovernorBravoDelegator__factory");
var CompoundGovernorBravoDelegate__factory_1 = require("../../typechain/factories/CompoundGovernorBravoDelegate__factory");
var CompoundTimelock__factory_1 = require("../../typechain/factories/CompoundTimelock__factory");
var Comptroller__factory_1 = require("../../typechain/factories/Comptroller__factory");
var PriceOracleProxy__factory_1 = require("../../typechain/factories/PriceOracleProxy__factory");
var Unitroller__factory_1 = require("../../typechain/factories/Unitroller__factory");
var WETH9__factory_1 = require("../../typechain/factories/WETH9__factory");
var WhitePaperInterestRateModel__factory_1 = require("../../typechain/factories/WhitePaperInterestRateModel__factory");
var LendingPoolAddressesProvider__factory_1 = require("../../typechain/factories/LendingPoolAddressesProvider__factory");
var AaveTokenV2Mintable__factory_1 = require("../../typechain/factories/AaveTokenV2Mintable__factory");
var AavePropositionPower__factory_1 = require("../../typechain/factories/AavePropositionPower__factory");
var AaveProtoGovernance__factory_1 = require("../../typechain/factories/AaveProtoGovernance__factory");
var AssetVotingWeightProvider__factory_1 = require("../../typechain/factories/AssetVotingWeightProvider__factory");
var LendingPoolCore__factory_1 = require("../../typechain/factories/LendingPoolCore__factory");
var CoreLibrary__factory_1 = require("../../typechain/factories/CoreLibrary__factory");
var GovernanceParamsProvider__factory_1 = require("../../typechain/factories/GovernanceParamsProvider__factory");
var LendingPool__factory_1 = require("../../typechain/factories/LendingPool__factory");
var DefaultReserveInterestRateStrategy__factory_1 = require("../../typechain/factories/DefaultReserveInterestRateStrategy__factory");
var LendingPoolConfigurator__factory_1 = require("../../typechain/factories/LendingPoolConfigurator__factory");
var LendingRateOracle__factory_1 = require("../../typechain/factories/LendingRateOracle__factory");
var LendingPoolDataProvider__factory_1 = require("../../typechain/factories/LendingPoolDataProvider__factory");
var LendToAaveMigrator__factory_1 = require("../../typechain/factories/LendToAaveMigrator__factory");
var CurvePoolERC20__factory_1 = require("../../typechain/factories/CurvePoolERC20__factory");
var Stableswap__factory_1 = require("../../typechain/factories/Stableswap__factory");
var CurveDeposit__factory_1 = require("../../typechain/factories/CurveDeposit__factory");
var CRVToken__factory_1 = require("../../typechain/factories/CRVToken__factory");
var GaugeController__factory_1 = require("../../typechain/factories/GaugeController__factory");
var LiquidityGaugeReward__factory_1 = require("../../typechain/factories/LiquidityGaugeReward__factory");
var Minter__factory_1 = require("../../typechain/factories/Minter__factory");
var LiquidityGauge__factory_1 = require("../../typechain/factories/LiquidityGauge__factory");
var StakingRewards__factory_1 = require("../../typechain/factories/StakingRewards__factory");
var Uni__factory_1 = require("../../typechain/factories/Uni__factory");
var UniswapGovernorAlpha__factory_1 = require("../../typechain/factories/UniswapGovernorAlpha__factory");
var UniswapTimelock__factory_1 = require("../../typechain/factories/UniswapTimelock__factory");
var UniswapV2Factory__factory_1 = require("../../typechain/factories/UniswapV2Factory__factory");
var UniswapV2Pair__factory_1 = require("../../typechain/factories/UniswapV2Pair__factory");
var UniswapV2Router02__factory_1 = require("../../typechain/factories/UniswapV2Router02__factory");
var BFactory__factory_1 = require("../../typechain/factories/BFactory__factory");
var BRegistry__factory_1 = require("../../typechain/factories/BRegistry__factory");
var ExchangeProxy__factory_1 = require("../../typechain/factories/ExchangeProxy__factory");
var DelegateRegistry__factory_1 = require("../../typechain/factories/DelegateRegistry__factory");
var Registry__factory_1 = require("../../typechain/factories/Registry__factory");
var Vault__factory_1 = require("../../typechain/factories/Vault__factory");
var DMMFactory__factory_1 = require("../../typechain/factories/DMMFactory__factory");
var DMMRouter02__factory_1 = require("../../typechain/factories/DMMRouter02__factory");
var DMMPool__factory_1 = require("../../typechain/factories/DMMPool__factory");
var KyberNetworkTokenV2__factory_1 = require("../../typechain/factories/KyberNetworkTokenV2__factory");
var TokenSwap__factory_1 = require("../../typechain/factories/TokenSwap__factory");
var SingularityNetToken__factory_1 = require("../../typechain/factories/SingularityNetToken__factory");
var UniswapV3Factory__factory_1 = require("../../typechain/factories/UniswapV3Factory__factory");
var UniswapV3Pool__factory_1 = require("../../typechain/factories/UniswapV3Pool__factory");
var SwapRouter__factory_1 = require("../../typechain/factories/SwapRouter__factory");
var NonfungiblePositionManager__factory_1 = require("../../typechain/factories/NonfungiblePositionManager__factory");
var Quoter__factory_1 = require("../../typechain/factories/Quoter__factory");
var NFTDescriptor__factory_1 = require("../../typechain/factories/NFTDescriptor__factory");
var AaveV2LendingPool__factory_1 = require("../../typechain/factories/AaveV2LendingPool__factory");
var AaveV2LendingPoolAddressesProvider__factory_1 = require("../../typechain/factories/AaveV2LendingPoolAddressesProvider__factory");
var AaveV2ProtocolDataProvider__factory_1 = require("../../typechain/factories/AaveV2ProtocolDataProvider__factory");
var AaveV2LendingPoolConfigurator__factory_1 = require("../../typechain/factories/AaveV2LendingPoolConfigurator__factory");
var AaveV2LendingPoolCollateralManager__factory_1 = require("../../typechain/factories/AaveV2LendingPoolCollateralManager__factory");
var AaveV2DefaultReserveInterestRateStrategy__factory_1 = require("../../typechain/factories/AaveV2DefaultReserveInterestRateStrategy__factory");
var AaveV2AToken__factory_1 = require("../../typechain/factories/AaveV2AToken__factory");
var AaveV2StableDebtToken__factory_1 = require("../../typechain/factories/AaveV2StableDebtToken__factory");
var AaveV2VariableDebtToken__factory_1 = require("../../typechain/factories/AaveV2VariableDebtToken__factory");
var AaveV2StakedTokenIncentivesController__factory_1 = require("../../typechain/factories/AaveV2StakedTokenIncentivesController__factory");
var GenericLogic__factory_1 = require("../../typechain/factories/GenericLogic__factory");
var ValidationLogic__factory_1 = require("../../typechain/factories/ValidationLogic__factory");
var ReserveLogic__factory_1 = require("../../typechain/factories/ReserveLogic__factory");
var AaveV2LendingRateOracle__factory_1 = require("../../typechain/factories/AaveV2LendingRateOracle__factory");
var AaveV2Oracle__factory_1 = require("../../typechain/factories/AaveV2Oracle__factory");
var AaveV2PriceOracle__factory_1 = require("../../typechain/factories/AaveV2PriceOracle__factory");
var AaveGovernanceV2__factory_1 = require("../../typechain/factories/AaveGovernanceV2__factory");
var Executor__factory_1 = require("../../typechain/factories/Executor__factory");
var GovernanceStrategy__factory_1 = require("../../typechain/factories/GovernanceStrategy__factory");
var PerpV2ClearingHouse__factory_1 = require("../../typechain/factories/PerpV2ClearingHouse__factory");
var PerpV2MarketRegistry__factory_1 = require("../../typechain/factories/PerpV2MarketRegistry__factory");
var PerpV2OrderBook__factory_1 = require("../../typechain/factories/PerpV2OrderBook__factory");
var PerpV2Quoter__factory_1 = require("../../typechain/factories/PerpV2Quoter__factory");
var PerpV2QuoteToken__factory_1 = require("../../typechain/factories/PerpV2QuoteToken__factory");
var PerpV2Vault__factory_1 = require("../../typechain/factories/PerpV2Vault__factory");
var PerpV2TestAggregatorV3__factory_1 = require("../../typechain/factories/PerpV2TestAggregatorV3__factory");
var PerpV2ChainlinkPriceFeed__factory_1 = require("../../typechain/factories/PerpV2ChainlinkPriceFeed__factory");
var PerpV2BaseToken__factory_1 = require("../../typechain/factories/PerpV2BaseToken__factory");
var PerpV2ClearingHouseConfig__factory_1 = require("../../typechain/factories/PerpV2ClearingHouseConfig__factory");
var PerpV2InsuranceFund__factory_1 = require("../../typechain/factories/PerpV2InsuranceFund__factory");
var PerpV2AccountBalance__factory_1 = require("../../typechain/factories/PerpV2AccountBalance__factory");
var PerpV2Exchange__factory_1 = require("../../typechain/factories/PerpV2Exchange__factory");
var DeployExternalContracts = /** @class */ (function () {
    function DeployExternalContracts(deployerSigner) {
        this._deployerSigner = deployerSigner;
    }
    // COMPOUND
    DeployExternalContracts.prototype.deployComp = function (_account) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Comp__factory_1.Comp__factory(this._deployerSigner).deploy(_account)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCompoundTimelock = function (_admin, _delay) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundTimelock__factory_1.CompoundTimelock__factory(this._deployerSigner).deploy(_admin, _delay)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCompoundGovernorAlpha = function (_timelock, _comp, _guardian) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundGovernorAlpha__factory_1.CompoundGovernorAlpha__factory(this._deployerSigner).deploy(_timelock, _comp, _guardian)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCompoundGovernorBravoDelegate = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundGovernorBravoDelegate__factory_1.CompoundGovernorBravoDelegate__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCompoundGovernorBravoDelegator = function (timelock, comp, admin, implementation, votingPeriod, votingDelay, proposalThreshold) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundGovernorBravoDelegator__factory_1.CompoundGovernorBravoDelegator__factory(this._deployerSigner).deploy(timelock, comp, admin, implementation, votingPeriod, votingDelay, proposalThreshold)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCeRc20 = function (underlying, comptroller, interestRateModel, initialExchangeRateMantissa, name, symbol, decimals) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CERc20__factory_1.CERc20__factory(this._deployerSigner).deploy(underlying, comptroller, interestRateModel, initialExchangeRateMantissa, name, symbol, decimals)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCEther = function (comptroller, interestRateModel, initialExchangeRateMantissa, name, symbol, decimals) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CEther__factory_1.CEther__factory(this._deployerSigner).deploy(comptroller, interestRateModel, initialExchangeRateMantissa, name, symbol, decimals)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCompoundPriceOracleMock = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CompoundPriceOracleMock__factory_1.CompoundPriceOracleMock__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPriceOracleProxy = function (guardian, v1PriceOracle, cEthAddress, cUsdcAddress, cSaiAddress, cDaiAddress, cUsdtAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PriceOracleProxy__factory_1.PriceOracleProxy__factory(this._deployerSigner).deploy(guardian, v1PriceOracle, cEthAddress, cUsdcAddress, cSaiAddress, cDaiAddress, cUsdtAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployComptroller = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Comptroller__factory_1.Comptroller__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployUnitroller = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Unitroller__factory_1.Unitroller__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployWhitePaperInterestRateModel = function (baseRate, multiplier) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new WhitePaperInterestRateModel__factory_1.WhitePaperInterestRateModel__factory(this._deployerSigner).deploy(baseRate, multiplier)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // WETH
    DeployExternalContracts.prototype.deployWETH = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new WETH9__factory_1.WETH9__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // AAVE
    DeployExternalContracts.prototype.deployAaveProtoGovernance = function (govParamsProvider) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveProtoGovernance__factory_1.AaveProtoGovernance__factory(this._deployerSigner).deploy(govParamsProvider)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployGovernanceParamsProvider = function (propositionPowerThreshold, propositionPower, assetVotingWeightProvider) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new GovernanceParamsProvider__factory_1.GovernanceParamsProvider__factory(this._deployerSigner).deploy(propositionPowerThreshold, propositionPower, assetVotingWeightProvider)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAavePropositionPower = function (name, symbol, decimals, council, cap) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AavePropositionPower__factory_1.AavePropositionPower__factory(this._deployerSigner).deploy(name, symbol, decimals, council, cap)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAssetVotingWeightProvider = function (assets, weights) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AssetVotingWeightProvider__factory_1.AssetVotingWeightProvider__factory(this._deployerSigner).deploy(assets, weights)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendingPoolAddressesProvider = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendingPoolAddressesProvider__factory_1.LendingPoolAddressesProvider__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCoreLibrary = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CoreLibrary__factory_1.CoreLibrary__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendingPoolCore = function (coreLibraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var lendingPoolCoreLibraryAddresses;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        lendingPoolCoreLibraryAddresses = {
                            __CoreLibrary___________________________: coreLibraryAddress,
                        };
                        return [4 /*yield*/, new LendingPoolCore__factory_1.LendingPoolCore__factory(lendingPoolCoreLibraryAddresses, this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendingPool = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendingPool__factory_1.LendingPool__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendingPoolConfigurator = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendingPoolConfigurator__factory_1.LendingPoolConfigurator__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployDefaultReserveInterestRateStrategy = function (_reserve, _AddressProvider, _baseVariableBorrowRate, _variableRateSlope1, _variableRateSlope2, _stableRateSlope1, _stableRateSlope2) {
        if (_baseVariableBorrowRate === void 0) { _baseVariableBorrowRate = (0, common_1.ether)(1); }
        if (_variableRateSlope1 === void 0) { _variableRateSlope1 = (0, common_1.ether)(1); }
        if (_variableRateSlope2 === void 0) { _variableRateSlope2 = (0, common_1.ether)(1); }
        if (_stableRateSlope1 === void 0) { _stableRateSlope1 = (0, common_1.ether)(1); }
        if (_stableRateSlope2 === void 0) { _stableRateSlope2 = (0, common_1.ether)(1); }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DefaultReserveInterestRateStrategy__factory_1.DefaultReserveInterestRateStrategy__factory(this._deployerSigner).deploy(_reserve, _AddressProvider, _baseVariableBorrowRate, _variableRateSlope1, _variableRateSlope2, _stableRateSlope1, _stableRateSlope2)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendingRateOracle = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendingRateOracle__factory_1.LendingRateOracle__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendingPoolDataProvider = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendingPoolDataProvider__factory_1.LendingPoolDataProvider__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLendToAaveMigrator = function (_aaveToken, _lendToken, _aaveLendRatio) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendToAaveMigrator__factory_1.LendToAaveMigrator__factory(this._deployerSigner).deploy(_aaveToken, _lendToken, _aaveLendRatio)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.getLendToAaveMigrator = function (lendToAaveMigratorAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LendToAaveMigrator__factory_1.LendToAaveMigrator__factory(this._deployerSigner).attach(lendToAaveMigratorAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // AAVE V2
    DeployExternalContracts.prototype.deployAaveV2LendingPoolAddressesProvider = function (marketId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2LendingPoolAddressesProvider__factory_1.AaveV2LendingPoolAddressesProvider__factory(this._deployerSigner).deploy(marketId)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2LendingPool = function (validationLogicAddress, reserveLogicAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new AaveV2LendingPool__factory_1.AaveV2LendingPool__factory((_a = {},
                            _a["__$de8c0cf1a7d7c36c802af9a64fb9d86036$__"] = validationLogicAddress,
                            _a["__$22cd43a9dda9ce44e9b92ba393b88fb9ac$__"] = reserveLogicAddress,
                            _a), this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2LendingPoolConfigurator = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2LendingPoolConfigurator__factory_1.AaveV2LendingPoolConfigurator__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2LendingPoolCollateralManager = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2LendingPoolCollateralManager__factory_1.AaveV2LendingPoolCollateralManager__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2LendingRateOracle = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2LendingRateOracle__factory_1.AaveV2LendingRateOracle__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2Oracle = function (assets, sources, fallBackOracle, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2Oracle__factory_1.AaveV2Oracle__factory(this._deployerSigner).deploy(assets, sources, fallBackOracle, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2PriceOracle = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2PriceOracle__factory_1.AaveV2PriceOracle__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2DefaultReserveInterestRateStrategy = function (_AddressesProvider, _optimalUtilizationRate, _baseVariableBorrowRate, _variableRateSlope1, _variableRateSlope2, _stableRateSlope1, _stableRateSlope2) {
        if (_optimalUtilizationRate === void 0) { _optimalUtilizationRate = (0, common_1.ether)(1); }
        if (_baseVariableBorrowRate === void 0) { _baseVariableBorrowRate = (0, common_1.ether)(1); }
        if (_variableRateSlope1 === void 0) { _variableRateSlope1 = (0, common_1.ether)(1); }
        if (_variableRateSlope2 === void 0) { _variableRateSlope2 = (0, common_1.ether)(1); }
        if (_stableRateSlope1 === void 0) { _stableRateSlope1 = (0, common_1.ether)(1); }
        if (_stableRateSlope2 === void 0) { _stableRateSlope2 = (0, common_1.ether)(1); }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2DefaultReserveInterestRateStrategy__factory_1.AaveV2DefaultReserveInterestRateStrategy__factory(this._deployerSigner).deploy(_AddressesProvider, _optimalUtilizationRate, _baseVariableBorrowRate, _variableRateSlope1, _variableRateSlope2, _stableRateSlope1, _stableRateSlope2)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2ProtocolDataProvider = function (addressProvider) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2ProtocolDataProvider__factory_1.AaveV2ProtocolDataProvider__factory(this._deployerSigner).deploy(addressProvider)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2AToken = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2AToken__factory_1.AaveV2AToken__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2StableDebtToken = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2StableDebtToken__factory_1.AaveV2StableDebtToken__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2VariableDebtToken = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2VariableDebtToken__factory_1.AaveV2VariableDebtToken__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveV2StakedTokenIncentivesController = function (stakeToken, emissionManager) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveV2StakedTokenIncentivesController__factory_1.AaveV2StakedTokenIncentivesController__factory(this._deployerSigner).deploy(stakeToken, emissionManager)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.getForkedAaveLendingPoolAddressesProvider = function (_mainnetAddressesProvider) {
        return AaveV2LendingPoolAddressesProvider__factory_1.AaveV2LendingPoolAddressesProvider__factory.connect(_mainnetAddressesProvider, this._deployerSigner);
    };
    DeployExternalContracts.prototype.getForkedAaveV2ProtocolDataProvider = function (_mainnetAddressesProvider) {
        return AaveV2ProtocolDataProvider__factory_1.AaveV2ProtocolDataProvider__factory.connect(_mainnetAddressesProvider, this._deployerSigner);
    };
    // AAVE V2 LIBRARIES
    DeployExternalContracts.prototype.deployGeneralLogic = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new GenericLogic__factory_1.GenericLogic__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployValidationLogic = function (genericLogicLibraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new ValidationLogic__factory_1.ValidationLogic__factory((_a = {},
                            _a["__$52a8a86ab43135662ff256bbc95497e8e3$__"] = genericLogicLibraryAddress,
                            _a), this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployReserveLogic = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new ReserveLogic__factory_1.ReserveLogic__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // AAVE V2 GOVERNANCE
    DeployExternalContracts.prototype.deployAaveGovernanceV2 = function (_governanceStrategy, _votingDelay, _guardian, _executors) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveGovernanceV2__factory_1.AaveGovernanceV2__factory(this._deployerSigner).deploy(_governanceStrategy, _votingDelay, _guardian, _executors)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployExecutor = function (_admin, _delay, _gracePeriod, _minimumDelay, _maximumDelay, _propositionThreshold, _voteDuration, _voteDifferential, _minmumQuorum) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Executor__factory_1.Executor__factory(this._deployerSigner).deploy(_admin, _delay, _gracePeriod, _minimumDelay, _maximumDelay, _propositionThreshold, _voteDuration, _voteDifferential, _minmumQuorum)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployGovernanceStrategy = function (_aave, _stkaave) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new GovernanceStrategy__factory_1.GovernanceStrategy__factory(this._deployerSigner).deploy(_aave, _stkaave)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployAaveTokenV2Mintable = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AaveTokenV2Mintable__factory_1.AaveTokenV2Mintable__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // Curve
    DeployExternalContracts.prototype.deployCurveDeposit = function (_coins, _underlying_coins, _curve, _token) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CurveDeposit__factory_1.CurveDeposit__factory(this._deployerSigner).deploy(_coins, _underlying_coins, _curve, _token)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCurvePoolERC20 = function (_name, _symbol, _decimals, _supply) {
        if (_decimals === void 0) { _decimals = 18; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CurvePoolERC20__factory_1.CurvePoolERC20__factory(this._deployerSigner).deploy(_name, _symbol, _decimals, _supply)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployStableswap = function (_coins, _underlying_coins, _pool_token, _aCoefficient, _fee) {
        if (_aCoefficient === void 0) { _aCoefficient = 1; }
        if (_fee === void 0) { _fee = 0; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Stableswap__factory_1.Stableswap__factory(this._deployerSigner).deploy(_coins, _underlying_coins, _pool_token, _aCoefficient, _fee)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployCrvToken = function (_name, _symbol, _decimals) {
        if (_decimals === void 0) { _decimals = 18; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CRVToken__factory_1.CRVToken__factory(this._deployerSigner).deploy(_name, _symbol, _decimals)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployGaugeController = function (_token, _voting_escrow) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new GaugeController__factory_1.GaugeController__factory(this._deployerSigner).deploy(_token, _voting_escrow)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLiquidityGaugeReward = function (_lpAddr, _minter, _reward_contract, _rewarded_token) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LiquidityGaugeReward__factory_1.LiquidityGaugeReward__factory(this._deployerSigner).deploy(_lpAddr, _minter, _reward_contract, _rewarded_token)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployLiquidityGauge = function (_lpAddr, _minter) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new LiquidityGauge__factory_1.LiquidityGauge__factory(this._deployerSigner).deploy(_lpAddr, _minter)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployMinter = function (_token, _controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Minter__factory_1.Minter__factory(this._deployerSigner).deploy(_token, _controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // Uniswap
    DeployExternalContracts.prototype.deployUni = function (_account, _minter, _mintingAllowedAfter) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Uni__factory_1.Uni__factory(this._deployerSigner).deploy(_account, _minter, _mintingAllowedAfter)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployUniswapTimelock = function (_admin, _delay) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapTimelock__factory_1.UniswapTimelock__factory(this._deployerSigner).deploy(_admin, _delay)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployUniswapGovernorAlpha = function (_timelock, _uni) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapGovernorAlpha__factory_1.UniswapGovernorAlpha__factory(this._deployerSigner).deploy(_timelock, _uni)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployUniswapV2Factory = function (_feeToSetter) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2Factory__factory_1.UniswapV2Factory__factory(this._deployerSigner).deploy(_feeToSetter)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployUniswapV2Router02 = function (_factory, _weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2Router02__factory_1.UniswapV2Router02__factory(this._deployerSigner).deploy(_factory, _weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.getForkedUniswapV2Router02 = function (_mainnetRouter) {
        return UniswapV2Router02__factory_1.UniswapV2Router02__factory.connect(_mainnetRouter, this._deployerSigner);
    };
    DeployExternalContracts.prototype.deployUniswapV2Pair = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV2Pair__factory_1.UniswapV2Pair__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployStakingRewards = function (_rewardsDistribution, _rewardsToken, _stakingToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new StakingRewards__factory_1.StakingRewards__factory(this._deployerSigner).deploy(_rewardsDistribution, _rewardsToken, _stakingToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployB__factory = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new BFactory__factory_1.BFactory__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployExchangeProxy = function (weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new ExchangeProxy__factory_1.ExchangeProxy__factory(this._deployerSigner).deploy(weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployBRegistry = function (factory) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new BRegistry__factory_1.BRegistry__factory(this._deployerSigner).deploy(factory)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployDelegateRegistry = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DelegateRegistry__factory_1.DelegateRegistry__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // YEARN
    DeployExternalContracts.prototype.deployVault = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Vault__factory_1.Vault__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.getVault = function (vaultAddress) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Vault__factory_1.Vault__factory(this._deployerSigner).attach(vaultAddress)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployRegistry = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Registry__factory_1.Registry__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // KYBER V3 DMM
    DeployExternalContracts.prototype.deployKyberNetworkTokenV2 = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new KyberNetworkTokenV2__factory_1.KyberNetworkTokenV2__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployDMMFactory = function (_feeToSetter) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DMMFactory__factory_1.DMMFactory__factory(this._deployerSigner).deploy(_feeToSetter)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployDMMRouter02 = function (_factory, _weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DMMRouter02__factory_1.DMMRouter02__factory(this._deployerSigner).deploy(_factory, _weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployDMMPool = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DMMPool__factory_1.DMMPool__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // AXIE-INFINITY
    DeployExternalContracts.prototype.deployTokenSwap = function (oldToken, newToken) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new TokenSwap__factory_1.TokenSwap__factory(this._deployerSigner).deploy(oldToken, newToken)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // Singularity Net
    DeployExternalContracts.prototype.deploySingularityNetToken = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new SingularityNetToken__factory_1.SingularityNetToken__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // Uniswap V3
    DeployExternalContracts.prototype.deployUniswapV3Factory = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV3Factory__factory_1.UniswapV3Factory__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deploySwapRouter = function (factory, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new SwapRouter__factory_1.SwapRouter__factory(this._deployerSigner).deploy(factory, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployNftPositionManager = function (factory, weth, nftDesc) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new NonfungiblePositionManager__factory_1.NonfungiblePositionManager__factory(this._deployerSigner).deploy(factory, weth, nftDesc)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployQuoter = function (factory, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new Quoter__factory_1.Quoter__factory(this._deployerSigner).deploy(factory, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployNFTDescriptor = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new NFTDescriptor__factory_1.NFTDescriptor__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.getUniswapV3PoolInstance = function (pool) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new UniswapV3Pool__factory_1.UniswapV3Pool__factory(this._deployerSigner).attach(pool)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // PerpV2
    DeployExternalContracts.prototype.deployPerpV2OrderBook = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2OrderBook__factory_1.PerpV2OrderBook__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2MarketRegistry = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2MarketRegistry__factory_1.PerpV2MarketRegistry__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2Quoter = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2Quoter__factory_1.PerpV2Quoter__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2QuoteToken = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2QuoteToken__factory_1.PerpV2QuoteToken__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2Vault = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2Vault__factory_1.PerpV2Vault__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2TestAggregatorV3 = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2TestAggregatorV3__factory_1.PerpV2TestAggregatorV3__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2ChainlinkPriceFeed = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2ChainlinkPriceFeed__factory_1.PerpV2ChainlinkPriceFeed__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2BaseToken = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2BaseToken__factory_1.PerpV2BaseToken__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2ClearingHouseConfig = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2ClearingHouseConfig__factory_1.PerpV2ClearingHouseConfig__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2InsuranceFund = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2InsuranceFund__factory_1.PerpV2InsuranceFund__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2AccountBalance = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2AccountBalance__factory_1.PerpV2AccountBalance__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2Exchange = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2Exchange__factory_1.PerpV2Exchange__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployExternalContracts.prototype.deployPerpV2ClearingHouse = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new PerpV2ClearingHouse__factory_1.PerpV2ClearingHouse__factory(this._deployerSigner).deploy()];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return DeployExternalContracts;
}());
exports.default = DeployExternalContracts;
