// @setprotocol/set-protocol-v2/dist/utils/deploys/deployModules.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var AaveLeverageModule__factory_1 = require("../../typechain/factories/AaveLeverageModule__factory");
var AirdropModule__factory_1 = require("../../typechain/factories/AirdropModule__factory");
var AmmModule__factory_1 = require("../../typechain/factories/AmmModule__factory");
var BasicIssuanceModule__factory_1 = require("../../typechain/factories/BasicIssuanceModule__factory");
var ClaimModule__factory_1 = require("../../typechain/factories/ClaimModule__factory");
var CompoundLeverageModule__factory_1 = require("../../typechain/factories/CompoundLeverageModule__factory");
var CustomOracleNavIssuanceModule__factory_1 = require("../../typechain/factories/CustomOracleNavIssuanceModule__factory");
var DebtIssuanceModule__factory_1 = require("../../typechain/factories/DebtIssuanceModule__factory");
var DebtIssuanceModuleV2__factory_1 = require("../../typechain/factories/DebtIssuanceModuleV2__factory");
var SlippageIssuanceModule__factory_1 = require("../../typechain/factories/SlippageIssuanceModule__factory");
var GeneralIndexModule__factory_1 = require("../../typechain/factories/GeneralIndexModule__factory");
var GovernanceModule__factory_1 = require("../../typechain/factories/GovernanceModule__factory");
var IssuanceModule__factory_1 = require("../../typechain/factories/IssuanceModule__factory");
var NavIssuanceModule__factory_1 = require("../../typechain/factories/NavIssuanceModule__factory");
var PerpV2LeverageModule__factory_1 = require("../../typechain/factories/PerpV2LeverageModule__factory");
var SingleIndexModule__factory_1 = require("../../typechain/factories/SingleIndexModule__factory");
var StakingModule__factory_1 = require("../../typechain/factories/StakingModule__factory");
var StreamingFeeModule__factory_1 = require("../../typechain/factories/StreamingFeeModule__factory");
var TradeModule__factory_1 = require("../../typechain/factories/TradeModule__factory");
var WrapModule__factory_1 = require("../../typechain/factories/WrapModule__factory");
var WrapModuleV2__factory_1 = require("../../typechain/factories/WrapModuleV2__factory");
var DeployModules = /** @class */ (function () {
    function DeployModules(deployerSigner) {
        this._deployerSigner = deployerSigner;
    }
    DeployModules.prototype.deployBasicIssuanceModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new BasicIssuanceModule__factory_1.BasicIssuanceModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployIssuanceModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new IssuanceModule__factory_1.IssuanceModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployDebtIssuanceModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DebtIssuanceModule__factory_1.DebtIssuanceModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployDebtIssuanceModuleV2 = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new DebtIssuanceModuleV2__factory_1.DebtIssuanceModuleV2__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deploySlippageIssuanceModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new SlippageIssuanceModule__factory_1.SlippageIssuanceModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployAmmModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AmmModule__factory_1.AmmModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.getBasicIssuanceModule = function (basicIssuanceModule) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new BasicIssuanceModule__factory_1.BasicIssuanceModule__factory(this._deployerSigner).attach(basicIssuanceModule)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployStreamingFeeModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new StreamingFeeModule__factory_1.StreamingFeeModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.getStreamingFeeModule = function (streamingFeeModule) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new StreamingFeeModule__factory_1.StreamingFeeModule__factory(this._deployerSigner).attach(streamingFeeModule)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployAirdropModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new AirdropModule__factory_1.AirdropModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployNavIssuanceModule = function (controller, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new NavIssuanceModule__factory_1.NavIssuanceModule__factory(this._deployerSigner).deploy(controller, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployTradeModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new TradeModule__factory_1.TradeModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployWrapModule = function (controller, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new WrapModule__factory_1.WrapModule__factory(this._deployerSigner).deploy(controller, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployClaimModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new ClaimModule__factory_1.ClaimModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployStakingModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new StakingModule__factory_1.StakingModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployCustomOracleNavIssuanceModule = function (controller, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new CustomOracleNavIssuanceModule__factory_1.CustomOracleNavIssuanceModule__factory(this._deployerSigner).deploy(controller, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.getNavIssuanceModule = function (navIssuanceModule) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new NavIssuanceModule__factory_1.NavIssuanceModule__factory(this._deployerSigner).attach(navIssuanceModule)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deploySingleIndexModule = function (controller, weth, uniswapRouter, sushiswapRouter, balancerProxy) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new SingleIndexModule__factory_1.SingleIndexModule__factory(this._deployerSigner).deploy(controller, weth, uniswapRouter, sushiswapRouter, balancerProxy)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployGeneralIndexModule = function (controller, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new GeneralIndexModule__factory_1.GeneralIndexModule__factory(this._deployerSigner).deploy(controller, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployGovernanceModule = function (controller) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new GovernanceModule__factory_1.GovernanceModule__factory(this._deployerSigner).deploy(controller)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployCompoundLeverageModule = function (controller, compToken, comptroller, cEth, weth, libraryName, libraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new CompoundLeverageModule__factory_1.CompoundLeverageModule__factory((_a = {},
                            _a[libraryName] = libraryAddress,
                            _a), this._deployerSigner).deploy(controller, compToken, comptroller, cEth, weth)];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployAaveLeverageModule = function (controller, lendingPoolAddressesProvider, libraryName, libraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new AaveLeverageModule__factory_1.AaveLeverageModule__factory((_a = {},
                            _a[libraryName] = libraryAddress,
                            _a), this._deployerSigner).deploy(controller, lendingPoolAddressesProvider)];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployWrapModuleV2 = function (controller, weth) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, new WrapModuleV2__factory_1.WrapModuleV2__factory(this._deployerSigner).deploy(controller, weth)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    DeployModules.prototype.deployPerpV2LeverageModule = function (controller, perpVault, perpQuoter, perpMarketRegistry, libraryName, libraryAddress) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0: return [4 /*yield*/, new PerpV2LeverageModule__factory_1.PerpV2LeverageModule__factory((_a = {},
                            _a[libraryName] = libraryAddress,
                            _a), this._deployerSigner).deploy(controller, perpVault, perpQuoter, perpMarketRegistry)];
                    case 1: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    return DeployModules;
}());
exports.default = DeployModules;
