// @setprotocol/set-protocol-v2/dist/utils/deploys/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var deployLibraries_1 = __importDefault(require("./deployLibraries"));
var deployMocks_1 = __importDefault(require("./deployMocks"));
var deployModules_1 = __importDefault(require("./deployModules"));
var deployCoreContracts_1 = __importDefault(require("./deployCoreContracts"));
var deployExternal_1 = __importDefault(require("./deployExternal"));
var deployAdapters_1 = __importDefault(require("./deployAdapters"));
var deployViewers_1 = __importDefault(require("./deployViewers"));
var deployProduct_1 = __importDefault(require("./deployProduct"));
var deployOracles_1 = __importDefault(require("./deployOracles"));
var DeployHelper = /** @class */ (function () {
    function DeployHelper(deployerSigner) {
        this.libraries = new deployLibraries_1.default(deployerSigner);
        this.mocks = new deployMocks_1.default(deployerSigner);
        this.modules = new deployModules_1.default(deployerSigner);
        this.core = new deployCoreContracts_1.default(deployerSigner);
        this.external = new deployExternal_1.default(deployerSigner);
        this.adapters = new deployAdapters_1.default(deployerSigner);
        this.viewers = new deployViewers_1.default(deployerSigner);
        this.product = new deployProduct_1.default(deployerSigner);
        this.oracles = new deployOracles_1.default(deployerSigner);
    }
    return DeployHelper;
}());
exports.default = DeployHelper;
