// @setprotocol/set-protocol-v2/dist/typechain/factories/ForceFunderMock__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForceFunderMock__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "destination",
                type: "address",
            },
        ],
        name: "fund",
        outputs: [],
        stateMutability: "payable",
        type: "function",
    },
];
var _bytecode = "0x6080604052348015600f57600080fd5b5060888061001e6000396000f3fe608060405260043610601c5760003560e01c806323024408146021575b600080fd5b604460048036036020811015603557600080fd5b50356001600160a01b03166046565b005b806001600160a01b0316fffea2646970667358221220cdb7aeb56523c5394cc8ee1cb71debcf2f13a6f4e82000ae13994a0dc32908d164736f6c634300060a0033";
var ForceFunderMock__factory = /** @class */ (function (_super) {
    __extends(ForceFunderMock__factory, _super);
    function ForceFunderMock__factory(signer) {
        return _super.call(this, _abi, _bytecode, signer) || this;
    }
    ForceFunderMock__factory.prototype.deploy = function (overrides) {
        return _super.prototype.deploy.call(this, overrides || {});
    };
    ForceFunderMock__factory.prototype.getDeployTransaction = function (overrides) {
        return _super.prototype.getDeployTransaction.call(this, overrides || {});
    };
    ForceFunderMock__factory.prototype.attach = function (address) {
        return _super.prototype.attach.call(this, address);
    };
    ForceFunderMock__factory.prototype.connect = function (signer) {
        return _super.prototype.connect.call(this, signer);
    };
    ForceFunderMock__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    ForceFunderMock__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    ForceFunderMock__factory.bytecode = _bytecode;
    ForceFunderMock__factory.abi = _abi;
    return ForceFunderMock__factory;
}(ethers_1.ContractFactory));
exports.ForceFunderMock__factory = ForceFunderMock__factory;
