// @setprotocol/set-protocol-v2/dist/typechain/factories/IAaveLendingPool__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IAaveLendingPool__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "core",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IAaveLendingPool__factory = /** @class */ (function () {
    function IAaveLendingPool__factory() {
    }
    IAaveLendingPool__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IAaveLendingPool__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IAaveLendingPool__factory.abi = _abi;
    return IAaveLendingPool__factory;
}());
exports.IAaveLendingPool__factory = IAaveLendingPool__factory;
