// @setprotocol/set-protocol-v2/dist/typechain/factories/IAaveLendingPoolCore__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IAaveLendingPoolCore__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_reserve",
                type: "address",
            },
        ],
        name: "getReserveATokenAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IAaveLendingPoolCore__factory = /** @class */ (function () {
    function IAaveLendingPoolCore__factory() {
    }
    IAaveLendingPoolCore__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IAaveLendingPoolCore__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IAaveLendingPoolCore__factory.abi = _abi;
    return IAaveLendingPoolCore__factory;
}());
exports.IAaveLendingPoolCore__factory = IAaveLendingPoolCore__factory;
