// @setprotocol/set-protocol-v2/dist/typechain/factories/IAmmAdapter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IAmmAdapter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_pool",
                type: "address",
            },
            {
                internalType: "address[]",
                name: "_components",
                type: "address[]",
            },
            {
                internalType: "uint256[]",
                name: "_maxTokensIn",
                type: "uint256[]",
            },
            {
                internalType: "uint256",
                name: "_minLiquidity",
                type: "uint256",
            },
        ],
        name: "getProvideLiquidityCalldata",
        outputs: [
            {
                internalType: "address",
                name: "_target",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_pool",
                type: "address",
            },
            {
                internalType: "address",
                name: "_component",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_maxTokenIn",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_minLiquidity",
                type: "uint256",
            },
        ],
        name: "getProvideLiquiditySingleAssetCalldata",
        outputs: [
            {
                internalType: "address",
                name: "_target",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_pool",
                type: "address",
            },
            {
                internalType: "address[]",
                name: "_components",
                type: "address[]",
            },
            {
                internalType: "uint256[]",
                name: "_minTokensOut",
                type: "uint256[]",
            },
            {
                internalType: "uint256",
                name: "_liquidity",
                type: "uint256",
            },
        ],
        name: "getRemoveLiquidityCalldata",
        outputs: [
            {
                internalType: "address",
                name: "_target",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_pool",
                type: "address",
            },
            {
                internalType: "address",
                name: "_component",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_minTokenOut",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_liquidity",
                type: "uint256",
            },
        ],
        name: "getRemoveLiquiditySingleAssetCalldata",
        outputs: [
            {
                internalType: "address",
                name: "_target",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_pool",
                type: "address",
            },
        ],
        name: "getSpenderAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_pool",
                type: "address",
            },
            {
                internalType: "address[]",
                name: "_components",
                type: "address[]",
            },
        ],
        name: "isValidPool",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IAmmAdapter__factory = /** @class */ (function () {
    function IAmmAdapter__factory() {
    }
    IAmmAdapter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IAmmAdapter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IAmmAdapter__factory.abi = _abi;
    return IAmmAdapter__factory;
}());
exports.IAmmAdapter__factory = IAmmAdapter__factory;
