// @setprotocol/set-protocol-v2/dist/typechain/factories/IBasicIssuanceModule__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IBasicIssuanceModule__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_quantity",
                type: "uint256",
            },
        ],
        name: "getRequiredComponentUnitsForIssue",
        outputs: [
            {
                internalType: "address[]",
                name: "",
                type: "address[]",
            },
            {
                internalType: "uint256[]",
                name: "",
                type: "uint256[]",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var IBasicIssuanceModule__factory = /** @class */ (function () {
    function IBasicIssuanceModule__factory() {
    }
    IBasicIssuanceModule__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IBasicIssuanceModule__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IBasicIssuanceModule__factory.abi = _abi;
    return IBasicIssuanceModule__factory;
}());
exports.IBasicIssuanceModule__factory = IBasicIssuanceModule__factory;
