// @setprotocol/set-protocol-v2/dist/typechain/factories/IClaimAdapter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IClaimAdapter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_rewardPool",
                type: "address",
            },
        ],
        name: "getClaimCallData",
        outputs: [
            {
                internalType: "address",
                name: "_subject",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_rewardPool",
                type: "address",
            },
        ],
        name: "getRewardsAmount",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_rewardPool",
                type: "address",
            },
        ],
        name: "getTokenAddress",
        outputs: [
            {
                internalType: "contract IERC20",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IClaimAdapter__factory = /** @class */ (function () {
    function IClaimAdapter__factory() {
    }
    IClaimAdapter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IClaimAdapter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IClaimAdapter__factory.abi = _abi;
    return IClaimAdapter__factory;
}());
exports.IClaimAdapter__factory = IClaimAdapter__factory;
