// @setprotocol/set-protocol-v2/dist/typechain/factories/IDebtIssuanceModule__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IDebtIssuanceModule__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
        ],
        name: "registerToIssuanceModule",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
        ],
        name: "unregisterFromIssuanceModule",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var IDebtIssuanceModule__factory = /** @class */ (function () {
    function IDebtIssuanceModule__factory() {
    }
    IDebtIssuanceModule__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IDebtIssuanceModule__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IDebtIssuanceModule__factory.abi = _abi;
    return IDebtIssuanceModule__factory;
}());
exports.IDebtIssuanceModule__factory = IDebtIssuanceModule__factory;
