// @setprotocol/set-protocol-v2/dist/typechain/factories/IExchangeAdapter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IExchangeAdapter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "getSpender",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_fromToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_toToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_toAddress",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_fromQuantity",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_minToQuantity",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_data",
                type: "bytes",
            },
        ],
        name: "getTradeCalldata",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IExchangeAdapter__factory = /** @class */ (function () {
    function IExchangeAdapter__factory() {
    }
    IExchangeAdapter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IExchangeAdapter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IExchangeAdapter__factory.abi = _abi;
    return IExchangeAdapter__factory;
}());
exports.IExchangeAdapter__factory = IExchangeAdapter__factory;
