// @setprotocol/set-protocol-v2/dist/typechain/factories/IGaugeController__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IGaugeController__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_gauge",
                type: "address",
            },
        ],
        name: "gauge_types",
        outputs: [
            {
                internalType: "int128",
                name: "",
                type: "int128",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IGaugeController__factory = /** @class */ (function () {
    function IGaugeController__factory() {
    }
    IGaugeController__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IGaugeController__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IGaugeController__factory.abi = _abi;
    return IGaugeController__factory;
}());
exports.IGaugeController__factory = IGaugeController__factory;
