// @setprotocol/set-protocol-v2/dist/typechain/factories/IIndexExchangeAdapter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IIndexExchangeAdapter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "getSpender",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_sourceToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_destinationToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_destinationAddress",
                type: "address",
            },
            {
                internalType: "bool",
                name: "_isSendTokenFixed",
                type: "bool",
            },
            {
                internalType: "uint256",
                name: "_sourceQuantity",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_destinationQuantity",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_data",
                type: "bytes",
            },
        ],
        name: "getTradeCalldata",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IIndexExchangeAdapter__factory = /** @class */ (function () {
    function IIndexExchangeAdapter__factory() {
    }
    IIndexExchangeAdapter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IIndexExchangeAdapter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IIndexExchangeAdapter__factory.abi = _abi;
    return IIndexExchangeAdapter__factory;
}());
exports.IIndexExchangeAdapter__factory = IIndexExchangeAdapter__factory;
