// @setprotocol/set-protocol-v2/dist/typechain/factories/IIntegrationRegistry__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IIntegrationRegistry__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_module",
                type: "address",
            },
            {
                internalType: "string",
                name: "_id",
                type: "string",
            },
            {
                internalType: "address",
                name: "_wrapper",
                type: "address",
            },
        ],
        name: "addIntegration",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_module",
                type: "address",
            },
            {
                internalType: "string",
                name: "_id",
                type: "string",
            },
        ],
        name: "getIntegrationAdapter",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_module",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "_id",
                type: "bytes32",
            },
        ],
        name: "getIntegrationAdapterWithHash",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_module",
                type: "address",
            },
            {
                internalType: "string",
                name: "_id",
                type: "string",
            },
        ],
        name: "isValidIntegration",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IIntegrationRegistry__factory = /** @class */ (function () {
    function IIntegrationRegistry__factory() {
    }
    IIntegrationRegistry__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IIntegrationRegistry__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IIntegrationRegistry__factory.abi = _abi;
    return IIntegrationRegistry__factory;
}());
exports.IIntegrationRegistry__factory = IIntegrationRegistry__factory;
