// @setprotocol/set-protocol-v2/dist/typechain/factories/IKyberNetworkProxy__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IKyberNetworkProxy__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_src",
                type: "address",
            },
            {
                internalType: "address",
                name: "_dest",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_srcQty",
                type: "uint256",
            },
        ],
        name: "getExpectedRate",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_src",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_srcAmount",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_dest",
                type: "address",
            },
            {
                internalType: "address",
                name: "_destAddress",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_maxDestAmount",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_minConversionRate",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_referalFeeAddress",
                type: "address",
            },
        ],
        name: "trade",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "payable",
        type: "function",
    },
];
var IKyberNetworkProxy__factory = /** @class */ (function () {
    function IKyberNetworkProxy__factory() {
    }
    IKyberNetworkProxy__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IKyberNetworkProxy__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IKyberNetworkProxy__factory.abi = _abi;
    return IKyberNetworkProxy__factory;
}());
exports.IKyberNetworkProxy__factory = IKyberNetworkProxy__factory;
