// @setprotocol/set-protocol-v2/dist/typechain/factories/IManagerIssuanceHook__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IManagerIssuanceHook__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_issueQuantity",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_sender",
                type: "address",
            },
            {
                internalType: "address",
                name: "_to",
                type: "address",
            },
        ],
        name: "invokePreIssueHook",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_redeemQuantity",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_sender",
                type: "address",
            },
            {
                internalType: "address",
                name: "_to",
                type: "address",
            },
        ],
        name: "invokePreRedeemHook",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var IManagerIssuanceHook__factory = /** @class */ (function () {
    function IManagerIssuanceHook__factory() {
    }
    IManagerIssuanceHook__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IManagerIssuanceHook__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IManagerIssuanceHook__factory.abi = _abi;
    return IManagerIssuanceHook__factory;
}());
exports.IManagerIssuanceHook__factory = IManagerIssuanceHook__factory;
