// @setprotocol/set-protocol-v2/dist/typechain/factories/IModuleIssuanceHook__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IModuleIssuanceHook__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_setTokenQuantity",
                type: "uint256",
            },
            {
                internalType: "contract IERC20",
                name: "_component",
                type: "address",
            },
            {
                internalType: "bool",
                name: "_isEquity",
                type: "bool",
            },
        ],
        name: "componentIssueHook",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_setTokenQuantity",
                type: "uint256",
            },
            {
                internalType: "contract IERC20",
                name: "_component",
                type: "address",
            },
            {
                internalType: "bool",
                name: "_isEquity",
                type: "bool",
            },
        ],
        name: "componentRedeemHook",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_setTokenQuantity",
                type: "uint256",
            },
        ],
        name: "moduleIssueHook",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_setTokenQuantity",
                type: "uint256",
            },
        ],
        name: "moduleRedeemHook",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var IModuleIssuanceHook__factory = /** @class */ (function () {
    function IModuleIssuanceHook__factory() {
    }
    IModuleIssuanceHook__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IModuleIssuanceHook__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IModuleIssuanceHook__factory.abi = _abi;
    return IModuleIssuanceHook__factory;
}());
exports.IModuleIssuanceHook__factory = IModuleIssuanceHook__factory;
