// @setprotocol/set-protocol-v2/dist/typechain/factories/INAVIssuanceModule__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.INAVIssuanceModule__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_reserveAsset",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_reserveAssetQuantity",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_minSetTokenReceiveQuantity",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_to",
                type: "address",
            },
        ],
        name: "issue",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_reserveAsset",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_setTokenQuantity",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "_minReserveReceiveQuantity",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_to",
                type: "address",
            },
        ],
        name: "redeem",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var INAVIssuanceModule__factory = /** @class */ (function () {
    function INAVIssuanceModule__factory() {
    }
    INAVIssuanceModule__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    INAVIssuanceModule__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    INAVIssuanceModule__factory.abi = _abi;
    return INAVIssuanceModule__factory;
}());
exports.INAVIssuanceModule__factory = INAVIssuanceModule__factory;
