// @setprotocol/set-protocol-v2/dist/typechain/factories/IPriceOracle__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPriceOracle__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_assetOne",
                type: "address",
            },
            {
                internalType: "address",
                name: "_assetTwo",
                type: "address",
            },
        ],
        name: "getPrice",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "masterQuoteAsset",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IPriceOracle__factory = /** @class */ (function () {
    function IPriceOracle__factory() {
    }
    IPriceOracle__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IPriceOracle__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IPriceOracle__factory.abi = _abi;
    return IPriceOracle__factory;
}());
exports.IPriceOracle__factory = IPriceOracle__factory;
