// @setprotocol/set-protocol-v2/dist/typechain/factories/IQuoter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IQuoter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "address",
                        name: "baseToken",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "isBaseToQuote",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "isExactInput",
                        type: "bool",
                    },
                    {
                        internalType: "uint256",
                        name: "amount",
                        type: "uint256",
                    },
                    {
                        internalType: "uint160",
                        name: "sqrtPriceLimitX96",
                        type: "uint160",
                    },
                ],
                internalType: "struct IQuoter.SwapParams",
                name: "params",
                type: "tuple",
            },
        ],
        name: "swap",
        outputs: [
            {
                components: [
                    {
                        internalType: "uint256",
                        name: "deltaAvailableBase",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "deltaAvailableQuote",
                        type: "uint256",
                    },
                    {
                        internalType: "int256",
                        name: "exchangedPositionSize",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "exchangedPositionNotional",
                        type: "int256",
                    },
                    {
                        internalType: "uint160",
                        name: "sqrtPriceX96",
                        type: "uint160",
                    },
                ],
                internalType: "struct IQuoter.SwapResponse",
                name: "response",
                type: "tuple",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var IQuoter__factory = /** @class */ (function () {
    function IQuoter__factory() {
    }
    IQuoter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IQuoter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IQuoter__factory.abi = _abi;
    return IQuoter__factory;
}());
exports.IQuoter__factory = IQuoter__factory;
