// @setprotocol/set-protocol-v2/dist/typechain/factories/ISetValuer__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ISetValuer__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_quoteAsset",
                type: "address",
            },
        ],
        name: "calculateSetTokenValuation",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var ISetValuer__factory = /** @class */ (function () {
    function ISetValuer__factory() {
    }
    ISetValuer__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    ISetValuer__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    ISetValuer__factory.abi = _abi;
    return ISetValuer__factory;
}());
exports.ISetValuer__factory = ISetValuer__factory;
