// @setprotocol/set-protocol-v2/dist/typechain/factories/IStakingAdapter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IStakingAdapter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_stakingContract",
                type: "address",
            },
        ],
        name: "getSpenderAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_stakingContract",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_notionalAmount",
                type: "uint256",
            },
        ],
        name: "getStakeCallData",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_stakingContract",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_notionalAmount",
                type: "uint256",
            },
        ],
        name: "getUnstakeCallData",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IStakingAdapter__factory = /** @class */ (function () {
    function IStakingAdapter__factory() {
    }
    IStakingAdapter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IStakingAdapter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IStakingAdapter__factory.abi = _abi;
    return IStakingAdapter__factory;
}());
exports.IStakingAdapter__factory = IStakingAdapter__factory;
