// @setprotocol/set-protocol-v2/dist/typechain/factories/IStreamingFeeModule__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IStreamingFeeModule__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
        ],
        name: "feeStates",
        outputs: [
            {
                components: [
                    {
                        internalType: "address",
                        name: "feeRecipient",
                        type: "address",
                    },
                    {
                        internalType: "uint256",
                        name: "maxStreamingFeePercentage",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "streamingFeePercentage",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "lastStreamingFeeTimestamp",
                        type: "uint256",
                    },
                ],
                internalType: "struct StreamingFeeModule.FeeState",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISetToken",
                name: "_setToken",
                type: "address",
            },
        ],
        name: "getFee",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IStreamingFeeModule__factory = /** @class */ (function () {
    function IStreamingFeeModule__factory() {
    }
    IStreamingFeeModule__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IStreamingFeeModule__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IStreamingFeeModule__factory.abi = _abi;
    return IStreamingFeeModule__factory;
}());
exports.IStreamingFeeModule__factory = IStreamingFeeModule__factory;
