// @setprotocol/set-protocol-v2/dist/typechain/factories/ISynthetixExchanger__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ISynthetixExchanger__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "from",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "sourceCurrencyKey",
                type: "bytes32",
            },
            {
                internalType: "uint256",
                name: "sourceAmount",
                type: "uint256",
            },
            {
                internalType: "bytes32",
                name: "destinationCurrencyKey",
                type: "bytes32",
            },
            {
                internalType: "address",
                name: "destinationAddress",
                type: "address",
            },
        ],
        name: "exchange",
        outputs: [
            {
                internalType: "uint256",
                name: "amountReceived",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "sourceAmount",
                type: "uint256",
            },
            {
                internalType: "bytes32",
                name: "sourceCurrencyKey",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "destinationCurrencyKey",
                type: "bytes32",
            },
        ],
        name: "getAmountsForExchange",
        outputs: [
            {
                internalType: "uint256",
                name: "amountReceived",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "fee",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "exchangeFeeRate",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var ISynthetixExchanger__factory = /** @class */ (function () {
    function ISynthetixExchanger__factory() {
    }
    ISynthetixExchanger__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    ISynthetixExchanger__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    ISynthetixExchanger__factory.abi = _abi;
    return ISynthetixExchanger__factory;
}());
exports.ISynthetixExchanger__factory = ISynthetixExchanger__factory;
