// @setprotocol/set-protocol-v2/dist/typechain/factories/IWrapV2Adapter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IWrapV2Adapter__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "ETH_TOKEN_ADDRESS",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_underlyingToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_wrappedToken",
                type: "address",
            },
        ],
        name: "getSpenderAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_underlyingToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_wrappedToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_wrappedTokenUnits",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_to",
                type: "address",
            },
            {
                internalType: "bytes",
                name: "_unwrapData",
                type: "bytes",
            },
        ],
        name: "getUnwrapCallData",
        outputs: [
            {
                internalType: "address",
                name: "_subject",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_underlyingToken",
                type: "address",
            },
            {
                internalType: "address",
                name: "_wrappedToken",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_underlyingUnits",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "_to",
                type: "address",
            },
            {
                internalType: "bytes",
                name: "_wrapData",
                type: "bytes",
            },
        ],
        name: "getWrapCallData",
        outputs: [
            {
                internalType: "address",
                name: "_subject",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_calldata",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IWrapV2Adapter__factory = /** @class */ (function () {
    function IWrapV2Adapter__factory() {
    }
    IWrapV2Adapter__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IWrapV2Adapter__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IWrapV2Adapter__factory.abi = _abi;
    return IWrapV2Adapter__factory;
}());
exports.IWrapV2Adapter__factory = IWrapV2Adapter__factory;
