// @setprotocol/set-protocol-v2/dist/typechain/factories/IYearnVault__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IYearnVault__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "pricePerShare",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "token",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IYearnVault__factory = /** @class */ (function () {
    function IYearnVault__factory() {
    }
    IYearnVault__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IYearnVault__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IYearnVault__factory.abi = _abi;
    return IYearnVault__factory;
}());
exports.IYearnVault__factory = IYearnVault__factory;
