// @setprotocol/set-protocol-v2/dist/typechain/factories/ModuleBase__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModuleBase__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "contract IController",
                name: "_controller",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "constructor",
    },
    {
        inputs: [],
        name: "controller",
        outputs: [
            {
                internalType: "contract IController",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "removeModule",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var ModuleBase__factory = /** @class */ (function () {
    function ModuleBase__factory() {
    }
    ModuleBase__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    ModuleBase__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    ModuleBase__factory.abi = _abi;
    return ModuleBase__factory;
}());
exports.ModuleBase__factory = ModuleBase__factory;
